src/platform_impl/macos/event_loop.rs

```diff
@@ -497,18 +466,17 @@ impl<T> EventLoopProxy<T> {
             CFRunLoopAddSource(rl, source, kCFRunLoopCommonModes);
             CFRunLoopWakeUp(rl);
 
-            EventLoopProxy { sender, source }
+            EventLoopProxy { user_wake_up, source }
         }
     }
 
-    pub fn send_event(&self, event: T) -> Result<(), EventLoopClosed<T>> {
-        self.sender.send(event).map_err(|mpsc::SendError(x)| EventLoopClosed(x))?;
+    pub fn wake_up(&self) {
+        self.user_wake_up.store(true, AtomicOrdering::Relaxed);
```

Comment:
I don't think there _is_ a correct ordering to use here, _technically_ the CPU could choose to execute all the code below, and wake up the main thread to process the event, before actually setting this flag (`SeqCst` is not strong enough to do this either, as it'd have to have been used in `CFRunLoopWakeUp` as well, which I'm not sure it has).

The proper way to do this would be to emit the wake-up event (or just set a flag) inside the `event_loop_proxy_handler` function above (which runs on the main thread). That's a bit of a larger rework though, the current impl will work fine for now (and the old code had the same problem), I'll change it myself at a later point.

Reply:
The only thing I care about is that the event is emitted only once and the wake up is not lost. Though, I think more intense sync should be on the user here...